### PR TITLE
remove paddle.fluid.dygraph.parallel.ParallelEnv

### PR DESCRIPTION
**Pull request by zhengqiwen1997**
### PR types

Others
### PR changes

APIs
### Describe

To clean the code in paddle.fluid, the useless codes should be removed and code still be used should move to a new position.

For function paddle.fluid.dygraph.parallel.ParallelEnv, since it is not used in version 2.x, should be removed.

For unittest code where this function is used, if it is aimed to test the old API, it can be removed directly. Otherwise, the paddle.fluid.dygraph.parallel.ParallelEnv should be replaced with paddle.distributed.ParallelEnv.

### CODE REVIEW

File: python/paddle/parallel.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    # [aliases] Compatible with old method names
+    local_rank = rank
+    nranks = world_size
+    dev_id = device_id
+
+
+# NOTE: [ Compatible ] Originally this class name is `Env`. The semantics of the old class names
+# are inaccurate and may confuse users, so replace it with `ParallelEnv`, but to be compatible
+# with the old examples, here still need to keep this name.
+Env = ParallelEnv
```
Comment (zoooo0820) [inline python/paddle/parallel.py]: `Env`is outdated and should be removed.

File: python/paddle/distributed/__init__.py [hunk 2]
```diff
@@ -72,7 +72,7 @@
 # (TODO: GhostScreaming) It needs migration of ParallelEnv. However,
 # it's hard to migrate APIs in paddle.fluid.dygraph.parallel completely.
 # It will be replaced later.
-from paddle.fluid.dygraph.parallel import ParallelEnv  # noqa: F401
+from ..parallel import ParallelEnv  # noqa: F401
```
Comment (zoooo0820) [inline python/paddle/distributed/__init__.py]: The note can be removed.

File: python/paddle/parallel.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+# limitations under the License.
+
+import os
+
+from paddle.fluid import core
+
+__all__ = []
+
+
+class ParallelEnv:
```
Comment (jeff41404) [inline python/paddle/parallel.py]: `ParallelEnv` is a API in `paddle.distributed`. It is inappropriate to put the implement here

**Review by jeff41404 — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**